### PR TITLE
Add a bunch of new scripts

### PR DESCRIPTION
**Pull request by 1ClickCoding**
SegoGithub for Godot Engine
ZugCodez (account no longer exists) for ModLoader64, Supermodel, and Duckstation (A few edits by me).
NotToDisturb for mGBA
mrcmunir for Vulkan Headers (test script)

### CODE REVIEW

**Comment by cobalt2727** [comment]
Whoops, GitHub mobile doesn't let you close with a comment.

This is a **_lot_** to put in one PR - can you break up each script into its own separate one?

File: scripts/games_and_emulators/duckstation.sh [hunk 1]
```diff
@@ -38,5 +38,56 @@ esac
...
+
+cd /usr/local/share/icons/hicolor/512x512/apps
+sudo wget -O duckstation-qt.png "https://raw.githubusercontent.com/$repository_username/L4T-Megascript/$repository_branch/assets/DuckStation/512x512.png"
+
+cd /usr/local/share/icons/hicolor/64x64/apps
+sudo wget -O duckstation-qt.png "https://raw.githubusercontent.com/$repository_username/L4T-Megascript/$repository_branch/assets/DuckStation/64x64.png"
+
+mkdir /usr/local/share/icons/hicolor/8x8 && mkdir /usr/local/share/icons/hicolor/8x8/apps && cd /usr/local/share/icons/hicolor/8x8/apps
+sudo wget -O duckstation-qt.png "https://raw.githubusercontent.com/$repository_username/L4T-Megascript/$repository_branch/assets/DuckStation/8x8.png"
+
```
Comment (theofficialgman) [inline scripts/games_and_emulators/duckstation.sh:79]: duck icons already exist in duckstation source code (see `duck.png`). We are not interested in unnecessary assets in the source repo of the L4T-Megascript

**Comment by theofficialgman** [comment]
To add to that. We have NO interest in any of the display manager scripts. Running alternative display manager with proper touchscreen mapping (and even internalizing xorg) often requires custom modification (see Switchroot/L4S discord for lightdm and sddm info). There is literally zero reason that we want users who have "no prior knowledge of Linux" (as our about page specifies our software is catered towards) to be using scripts that will leave their system in a broken state where they can't even enter a GUI.

File: megascript_apps.txt [hunk 2]
```diff
@@ -34,8 +34,14 @@ fn="SM64Port";;d="A native port of the classic game for the N64 (requires a ROM)
 
 fn="Citra";;d="3DS emulator";;sn="citra.sh";;f="games_and_emulators"
 
+fn="DuckStation";;d="DuckStation is a PS1 Emulator aiming for the best accuracy and game support.";;sn="duckstation.sh";;f="games_and_emulators"
+
+fn="DobieStation";;d="A dog-themed PS2 emulator.";;sn="dobiestation.sh";;f="games_and_emulators"
```
Comment (theofficialgman) [inline megascript_apps.txt:39]: `dobiestation.sh` is not in the GUI on purpose:
```
#I haven't tested this, but it looks pretty unoptimized. Mostly just here for the far future if speed improves
```

File: megascript_apps_fedora.txt [hunk 3]
```diff
@@ -33,9 +33,18 @@ fn="Metaforce";;d="An attempt to recreate the engine for Retro Studios' GameCube
 # fn="SM64Port";;d="A native port of the classic game for the N64 (requires a ROM)";;sn="SM64.sh";;f="games_and_emulators"
 
 fn="Citra";;d="3DS emulator";;sn="citra.sh";;f="games_and_emulators"
+<<<<<<< Updated upstream
+=======
+
+# fn="DuckStation";;d="DuckStation is a PS1 Emulator aiming for the best accuracy and game support.";;sn="duckstation.sh";;f="games_and_emulators"
+
+# fn="DobieStation";;d="A dog-themed PS2 emulator.";;sn="dobiestation.sh";;f="games_and_emulators"
+>>>>>>> Stashed changes
```
Comment (theofficialgman) [inline megascript_apps_fedora.txt:42]: git stash output that needs to be corrected or removed

File: megascript_apps_fedora.txt [hunk 4]
```diff
@@ -93,3 +104,19 @@ fn="Github CLI";;d="GitHub's official command line tool";;sn="github-cli.sh";;f=
...
+
+# fn="X Window Display Manager";;d="The X Window Display Manager (XDM) is the default display manager for the X Window System.";;sn="xdm.sh";;f="display_managers"
+
+# fn="LightDM";;d="LightDM is a cross-desktop display manager. It was built as a relatively light-weight and highly customizable alternative to GDM.";;sn="lightdm.sh";;f="display_managers"
+
+# fn="Simple Log in Manager";;d="Simple Login Manager (SLiM) is a graphical login manager for X11 that aims to be simple, fast and independent from the various desktop environments.";;sn="slim.sh";;f="display_managers"
+
+# fn="WINGs Display Manager";;d="WINGs Display Manager (WDM) is a a display manager created using the WINGs widget-set used in Window Maker.";;sn="wdm.sh";;f="display_managers"
+
+# fn="NoDM";;d="NoDM is a minimal display manager that simply logs in as a given user and starts an X session, without asking for username or password.";;sn="nodm.sh";;f="display_managers"
```
Comment (theofficialgman) [inline megascript_apps_fedora.txt:122]: adding these commented out to the fedora apps list when the base scripts have note even partial fedora support serves no purpose

the reason the other scripts exist commented out here is they have partial but not complete fedora support or they were in the initial copy from the ubuntu/debian list. there is no need to add additional commented out scripts here that have no support

File: scripts/development/godot.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
+if which godot3 > /dev/null 2>&1; then
```
Comment (theofficialgman) [inline scripts/development/godot.sh:1]: I expect developers that need godot to be able to follow the official godot documentation for the particular version of godot that they are targeting for there game or project. 

a buildscript for one specific version of godot does not really help anyone, I don't want to maintain it, and I can't adequately devote time to testing that the editor continues to function properly.

following the documentation is all that is necessary for developers looking to use godot https://docs.godotengine.org/en/stable/contributing/development/compiling/compiling_for_linuxbsd.html

File: scripts/vulkan-headers.sh [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+echo "Vulkan script started"
+
+
+sudo apt install cmake gcc -y || error "Could not install packages"
+cd /tmp || error "Could not move to /tmp folder"
+rm -rf ./Vulkan-Headers
+git clone https://github.com/KhronosGroup/Vulkan-Headers.git
```
Comment (theofficialgman) [inline scripts/vulkan-headers.sh:8]: vulkan headers are already provided via the launchpad ppa added on all L4T-Megascript users install
https://launchpad.net/~theofficialgman/+archive/ubuntu/gpu-tools/+packages?field.name_filter=&field.status_filter=published&field.series_filter=bionic

for developers that need even newer vulkan headers (even though L4T R32 only has a max of vulkan 1.2) they should be capable of compiling the headers themselves following official documentation.

File: scripts/games_and_emulators/duckstation.sh (hunk 1 above)
Comment (1ClickCoding) [reply scripts/games_and_emulators/duckstation.sh:79]: the source code doesn't have an scalable svg and the png is the wrong size.

File: megascript_apps.txt (hunk 2 above)
Comment (1ClickCoding) [reply megascript_apps.txt:39]: added a hashtag for dobiestation so it's not there.

File: megascript_apps_fedora.txt (hunk 3 above)
Comment (1ClickCoding) [reply megascript_apps_fedora.txt:42]: Didn't notice it was there.

File: scripts/vulkan-headers.sh (hunk 6 above)
Comment (1ClickCoding) [reply scripts/vulkan-headers.sh:8]: This isn't my script, I was just compiling/fixing scripts from other forks into 1 big repo

File: scripts/development/godot.sh (hunk 5 above)
Comment (1ClickCoding) [reply scripts/development/godot.sh:1]: Again, this isn't my script, This came from another pull request (the one from SegoGithub).

**Comment by 1ClickCoding** [comment]
We could use the display manager scripts for non-touchscreen ARM-based systems. (like Pi, Jetson, and Arduino.)

File: megascript_apps_fedora.txt (hunk 4 above)
Comment (1ClickCoding) [reply megascript_apps_fedora.txt:122]: I will remove these from the list.

File: scripts/development/godot.sh (hunk 5 above)
Comment (1ClickCoding) [reply scripts/development/godot.sh:1]: I will remove this script from the list tho.

**Pull request closed without merging**